Implement config validation (#2645)
skypilot-org/skypilot: docs/source/reference/config.rst

      # The OCID of the compartment to use for launching instances (optional).
      compartment_ocid: ocid1.compartment.oc1..aaaaaaaahr7aicqtodxmcfor6pbqn3hvsngpftozyxzqw36gj4kh3w3kkj4q
      # The image tag to use for launching general instances (optional).
      image_tag_general: skypilot:cpu-ubuntu-2004
      # The image tag to use for launching GPU instances (optional).
      image_tag_gpu: skypilot:gpu-ubuntu-2004

    ap-seoul-1:
      # The OCID of the subnet to use for instances (optional).
      vcn_subnet: ocid1.subnet.oc1.ap-seoul-1.aaaaaaaa5c6wndifsij6yfyfehmi3tazn6mvhhiewqmajzcrlryurnl7nuja

    us-ashburn-1:
      vcn_subnet: ocid1.subnet.oc1.iad.aaaaaaaafbj7i3aqc4ofjaapa5edakde6g4ea2yaslcsay32cthp7qo55pxa

  # Advanced Kubernetes configurations (optional).
  kubernetes:
    # The networking mode for accessing jump pods (optional).
    # This must be either: 'nodeport' or 'portforward'.
    #
    # nodeport: Exposes a service on a static port number on each Node, allowing external access to services using <NodeIP>:<NodePort>.
    # portforward: Forward a local port to a port on a Pod, typically used for debugging or temporary access purposes.

>     networking:  nodeport

Author: Done!

      # Only one element is allowed in this list, as GCP disallows multiple
      # specific_reservations in a single request.
      - projects/my-project/reservations/my-reservation

  # Advanced Kubernetes configurations (optional).
  kubernetes:

>     # The networking mode for accessing jump pods (optional).

Reviewer: ```suggestion
    # The networking mode for accessing SSH jump pod (optional).
```

>     # This must be either: 'nodeport' or 'portforward'.

Reviewer: ```suggestion
    # This must be either: 'nodeport' or 'portforward'. If not specified, defaults to 'portforward'.
```

    #

>     # nodeport: Exposes a service on a static port number on each Node, allowing external access to services using <NodeIP>:<NodePort>.

Reviewer: ```suggestion
    # nodeport: Exposes the jump pod SSH service on a static port number on each Node, allowing external access to using <NodeIP>:<NodePort>. Using this mode requires opening multiple ports on nodes in the Kubernetes cluster.
```

>     # portforward: Forward a local port to a port on a Pod, typically used for debugging or temporary access purposes.

Reviewer: ```suggestion
    # portforward: Uses `kubectl port-forward` to create a tunnel and directly access the jump pod SSH service in the Kubernetes cluster. Does not require opening ports the cluster nodes and is more secure. 'portforward' is used as default if 'networking' is not specified.
```

>     networking:  nodeport

Reviewer: There seem to be two spaces here :) Also, should we default to portforward here?
```suggestion
    networking: portforward
```
